homeassistant/components/bme280/sensor.py

```diff
@@ -4,132 +4,130 @@
-            int
-        ),
-        vol.Optional(CONF_T_STANDBY, default=DEFAULT_T_STANDBY): vol.Coerce(int),
-        vol.Optional(CONF_FILTER_MODE, default=DEFAULT_FILTER_MODE): vol.Coerce(int),
-        vol.Optional(CONF_DELTA_TEMP, default=DEFAULT_DELTA_TEMP): vol.Coerce(float),
-    }
-)
+SCAN_INTERVAL = timedelta(seconds=DEFAULT_SCAN_INTERVAL)
+
+
+def _set_scan_interval(interval: timedelta) -> None:
+    global SCAN_INTERVAL  # pylint: disable=global-statement
+    SCAN_INTERVAL = interval
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the BME280 sensor."""
-
+    if discovery_info is None:
+        return
     SENSOR_TYPES[SENSOR_TEMP][1] = hass.config.units.temperature_unit
-    name = config[CONF_NAME]
-    i2c_address = config[CONF_I2C_ADDRESS]
-
-    bus = smbus.SMBus(config[CONF_I2C_BUS])
-    sensor = await hass.async_add_executor_job(
-        partial(
-            BME280,
-            bus,
-            i2c_address,
-            osrs_t=config[CONF_OVERSAMPLING_TEMP],
-            osrs_p=config[CONF_OVERSAMPLING_PRES],
-            osrs_h=config[CONF_OVERSAMPLING_HUM],
-            mode=config[CONF_OPERATION_MODE],
-            t_sb=config[CONF_T_STANDBY],
-            filter_mode=config[CONF_FILTER_MODE],
-            delta_temp=config[CONF_DELTA_TEMP],
-            logger=_LOGGER,
+    sensor_conf = discovery_info[DOMAIN]
+    _set_scan_interval(sensor_conf[CONF_SCAN_INTERVAL])
+    name = sensor_conf[CONF_NAME]
+    if CONF_SPI_BUS in sensor_conf and CONF_SPI_DEV in sensor_conf:
+        spi_dev = sensor_conf[CONF_SPI_DEV]
+        spi_bus = sensor_conf[CONF_SPI_BUS]
+        _LOGGER.debug("BME280 sensor initialize at %s.%s", spi_bus, spi_dev)
+        sensor = await hass.async_add_executor_job(
+            partial(
+                BME280_spi,
+                t_mode=sensor_conf[CONF_OVERSAMPLING_TEMP],
+                p_mode=sensor_conf[CONF_OVERSAMPLING_PRES],
+                h_mode=sensor_conf[CONF_OVERSAMPLING_HUM],
+                standby=sensor_conf[CONF_T_STANDBY],
+                filter=sensor_conf[CONF_FILTER_MODE],
+                spi_bus=sensor_conf[CONF_SPI_BUS],
+                spi_dev=sensor_conf[CONF_SPI_DEV],
+            )
         )
-    )
-    if not sensor.sample_ok:
-        _LOGGER.error("BME280 sensor not detected at %s", i2c_address)
-        return False
-
-    sensor_handler = await hass.async_add_executor_job(BME280Handler, sensor)
-
-    dev = []
-    with suppress(KeyError):
-        for variable in config[CONF_MONITORED_CONDITIONS]:
-            dev.append(
-                BME280Sensor(sensor_handler, variable, SENSOR_TYPES[variable][1], name)
+        if not sensor.sample_ok:
+            _LOGGER.error("BME280 sensor not detected at %s.%s", spi_bus, spi_dev)
+            return
+        sensor.update()
+        sensor_handler = await hass.async_add_executor_job(BME280Handler, sensor, INTERFACE_SPI)
```

Comment:
We don't need to use the executor if there are no side effects when instantiating the handler.